binding.gyp

```diff
@@ -1,68 +1,123 @@
-                ],
-            },
-            'sources': [
-                'src/main.cc',
-                'src/tsp.cc',
-                'src/vrp.cc',
-            ],
-            'ldflags': [
-                '-Wl,-z,now'
-            ],
-            'conditions': [
-                ['OS == "linux"', {
-                    'cflags': [
-                        '<@(system_includes)'
-                    ]
-                }]
-            ],
+          'action_name': 'link_deps',
+          'inputs': ['./node_modules/.bin/mason-js'],
+          'outputs': ['./mason_packages/.link'],
+          'action': ['./node_modules/.bin/mason-js', 'link']
+        }
+      ]
+    },
+    {
+      # module_name and module_path are both variables passed by node-pre-gyp from package.json
+      'target_name': '<(module_name)', # sets the name of the binary file
+      'product_dir': '<(module_path)', # controls where the node binary file gets copied to (./lib/binding/module.node)
+      'type': 'loadable_module',
+      'dependencies': [ 'action_before_build' ],
+      # "make" only watches files specified here, and will sometimes cache these files after the first compile.
+      # This cache can sometimes cause confusing errors when removing/renaming/adding new files. 
+      # Running "make clean" helps to prevent this "mysterious error by cache" scenario
+      # This also is where the benefits of using a "glob" come into play...
+      # See: https://github.com/mapbox/node-cpp-skel/pull/44#discussion_r122050205
+      'link_settings': {
+          'libraries': [
+            '<(module_root_dir)/mason_packages/.link/lib/libortools.a',
+            '<(module_root_dir)/mason_packages/.link/lib/libprotobuf.a',
+            '<(module_root_dir)/mason_packages/.link/lib/libgflags.a',
+            '-lz',
+          ],
+      },
+      'sources': [
+          './src/main.cc',
+          './src/tsp.cc',
+          './src/vrp.cc',
+      ],
+      'ldflags': [
+        '-Wl,-z,now',
+      ],
+      'conditions': [
+        ['error_on_warnings == "true"', {
+            'cflags_cc' : [ '-Werror' ],
             'xcode_settings': {
-                'OTHER_LDFLAGS':[
-                  '-Wl,-bind_at_load'
-                ],
-                'OTHER_CPLUSPLUSFLAGS': [
-                    '<@(system_includes)'
-                ],
-                'GCC_ENABLE_CPP_RTTI': 'YES',
-                'GCC_ENABLE_CPP_EXCEPTIONS': 'YES',
-                'MACOSX_DEPLOYMENT_TARGET':'10.8',
-                'CLANG_CXX_LIBRARY': 'libc++',
-                'CLANG_CXX_LANGUAGE_STANDARD':'c++14',
-                'GCC_VERSION': 'com.apple.compilers.llvm.clang.1_0'
+              'OTHER_CPLUSPLUSFLAGS': [ '-Werror' ]
             }
-        },
-    ],
+        }]
+      ],
+      'cflags_cc': [
+          '-std=c++14',
+          '<@(system_includes)',
+          '<@(compiler_checks)'
+      ],
+      'xcode_settings': {
+        'OTHER_LDFLAGS':[
+          '-Wl,-bind_at_load'
```

Comment:
Why?

Reply:
This is necessary for dynamic libraries to load inside the nodejs environment.  The previous `binding.gyp` had the same setting (see up a few lines), it's just been moved to a new location.